[matchit] Add first draft port

daljit97 opened the pull request:
Add matchit, a library for pattern matching for C++17

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
I am still working on this PR

daljit97 committed:
Add first draft port for matchit

BowenFu commented:
@daljit97 maybe the tests failed because you did not run ./vcpkg x-add-version --all?

daljit97 committed:
Adjust formatting

daljit97 committed:
Add version for matchit

daljit97 commented:
@BowenFu you're right I forgot to that, but that doesn't seem the issue. The problem seems to be that the files matchitConfig.cmake and matchitConfigVersion.cmake aren't generated in the buildtree of vcpkg (they are when running cmake ordinarily).

BowenFu commented:
@daljit97 maybe the problem is that make install was not supported in v1.0.0, will add new version v1.0.1 to fix this.

Cheney-W commented:
If this is a header-only library, you can refer to [this file](https://github.com/microsoft/vcpkg/blob/master/ports/websocketpp/portfile.cmake) to modify your PR.

daljit97 committed:
Update matchit to v1.0.1

daljit97 committed:
Run vcpkg x-add-version for v1.0.1

daljit97 committed:
Fix typo in command for copying license file

daljit97 committed:
Remove installation of redudant files

daljit97 committed:
Add usage instructions

daljit97 committed:
Update license info in vcpkg.json

daljit97 commented:
@BowenFu thanks indeed that seemed to be the issue (I probably should've checked against the 1.0.0 tag instead of head). I've tested on Linux and Windows and the port seems to work (at least locally).

daljit97 committed:
FIX: revert license change info to libzippp

Previous commit incorrectly change the license for the port libzippp. This commit fixes that.

daljit97 committed:
Run vcpkg x-add-version

BowenFu commented on ports/matchit/portfile.cmake:
```diff
@@ -0,0 +1,23 @@
+    REPO BowenFu/matchit.cpp
+    REF v1.0.1
+    SHA512 60edc6a392f5629391fa9e3ff09b7b98a0a782919a066ad2999eabb58e60f38bd50e080037b1276c5bca986f81ca0dfff2914816d46458b7b4e1c947a6134169
+    HEAD_REF master
```
```suggestion
    HEAD_REF main
```

daljit97 committed:
Update ports/matchit/portfile.cmake

Change HEAD_REF to main

Co-authored-by: Bowen Fu <BowenFu@users.noreply.github.com>

daljit97 committed:
Run vcpkg xdg-version

daljit97 committed:
Add command for installing usage file

daljit97 committed:
Run vcpkg x-add-version

Cheney-W committed:
Update matchit.json

Cheney-W commented on ports/matchit/portfile.cmake:
```diff
@@ -0,0 +1,23 @@
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/lib")
+
+file(INSTALL "${CMAKE_CURRENT_LIST_DIR}/usage" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}")
+file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)
```
```suggestion
file(INSTALL "${SOURCE_PATH}/LICENSE" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)
```

daljit97 replied:
@Cheney-W Committed.

daljit97 committed:
Use quoted variables in portfile.cmake

Co-authored-by: Cheney Wang <38240633+Cheney-W@users.noreply.github.com>

daljit97 committed:
Run x-add-version

JavierMatosD commented:
Thank you!
